pkg/client/testing/core/fixture.go

```diff
@@ -209,41 +223,124 @@ func (o objects) Kind(kind unversioned.GroupVersionKind, name string) (runtime.O
 	return out, nil
 }
 
-func (o objects) Add(obj runtime.Object) error {
-	gvks, _, err := o.scheme.ObjectKinds(obj)
+func (r *registry) Add(obj runtime.Object) error {
+	return r.add(obj, false)
```

Comment:
Check for existence before starting, you should produce an "already exists" error.